contrib/pyln-proto/tests/test_zbase32.py

```diff
@@ -0,0 +1,106 @@
+import pytest
+import bitstring
+from pyln.proto import zbase32
+
+not_str_bytes = [1, dict(), None, 3.4, object()]
+messages = [b'this', b'is', b'a', b'split', b'message:', b'lightning', b'rocks']
+zbase_encoded_messages = [b'qtwg1ha', b'pf3o', b'cr', b'qpaga4mw', b'pi1zgh5bc71uw', b'ptwsq4dwp3wsh3a', b'qjzsg45u']
+not_zbase_encoded_messages = ['00', '[]', 'vv', '1234']
+
+
+def test_message_to_bitarray():
+    # This is an internal function and the input is supposed to be bytes. Not testing unexpected inputs.
+    dummy_messages = [b'a' * i for i in range(1, 6)]
+    for message in dummy_messages:
+        not_filled_barr = bitstring.Bits(message)
+        barr = zbase32._message_to_bitarray(message)
+        assert isinstance(barr, bitstring.ConstBitStream)
+
+        for i in range(len(barr)):
+            # Then first len(not_filled_barr) bits are equal
+            if i < len(not_filled_barr):
+                assert not_filled_barr.bin[i] == barr.bin[i]
+            # The remaining are zeros
+            else:
+                assert barr.bin[i] == '0'
+
+
+def test_bitarray_to_message():
+    # This is an internal function and the input is supposed to be BitArray. Not testing unexpected inputs.
+    dummy_messages = [b'b' * i for i in range(1, 6)]
+
+    for message in dummy_messages:
+        barr = bitstring.Bits(message)
+        not_filled_message = zbase32._bitarray_to_message(bitstring.BitArray(message))
+        assert isinstance(not_filled_message, bytes)
+
+        not_filled_barr = bitstring.Bits(not_filled_message)
+        for i in range(len(not_filled_barr)):
+            # Then first len(pre_filled_barr) bits are equal
+            if i < len(not_filled_barr):
+                assert not_filled_barr.bin[i] == barr.bin[i]
+            # The remaining are zeros
+            else:
+                assert barr.bin[i] == '0'
+
+
+def test_bitarray_to_u5():
+    # This is an internal function and the input is supposed to be ConstBitStream of length multiple of 5.
+    # Not testing unexpected inputs.
+    barrs = [zbase32._message_to_bitarray(message) for message in messages]
+
+    for barr in barrs:
+        u5 = zbase32._bitarray_to_u5(barr)
+        assert isinstance(u5, list)
+        assert all(x in range(0, 31) for x in u5)
+
+
+def test_u5_to_bitarray():
+    # This is an internal function and the input is supposed to be a list of ints 0-31. Not testing unexpected inputs.
+    u5s = [[0, 1, 2, 3, 4, 5], [15, 30, 24, 17, 8, 3, 21], [0], [28, 11]]
+
+    for u5 in u5s:
+        bitarray = zbase32._u5_to_bitarray(u5)
+        assert isinstance(bitarray, bitstring.BitArray)
+
+
+def test_is_zbase32_encoded():
+    for message in zbase_encoded_messages:
+        assert zbase32.is_zbase32_encoded(message)
+
+    for message in not_zbase_encoded_messages:
+        assert not zbase32.is_zbase32_encoded(message)
+
+
+def test_encode():
+    for i, message in enumerate(messages):
```

Comment:
You can iterate over both arrays `messages` and `zbase_encoded_messages` using `zip(messages, zbase_encoded_messages)`. This way you can just loop like this:

```
for mess, expected in zip(messages, zbase_encoded_messages):
     got = zbase32.encode(message)
     assert(got == expected)
```

Reply:
Alternatively you can also make the messages a `dict`, mapping from the raw message to the zbase32-encoded messages and then iterate over `messages.items()`.

Reply:
Yeah, indeed. That's way nicer.